src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -187,6 +227,204 @@ uct_gga_mlx5_rkey_resolve(uct_ib_mlx5_md_t *md, uct_rkey_t rkey)
 
 static UCS_CLASS_DECLARE_DELETE_FUNC(uct_gga_mlx5_iface_t, uct_iface_t);
 
+static ucs_status_t
+uct_gga_mlx5_iface_event_fd_get(uct_iface_h tl_iface, int *fd_p)
+{
+    uct_rc_mlx5_iface_common_t *iface =
+            ucs_derived_of(tl_iface, uct_rc_mlx5_iface_common_t);
+    uct_ib_mlx5_md_t *md UCS_V_UNUSED =
+            uct_ib_mlx5_iface_md(&iface->super.super);
+
+    ucs_assert(md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_CQ);
+    *fd_p = iface->cq_event_channel->fd;
+    return UCS_OK;
+}
+
+static ucs_status_t
+uct_gga_mlx5_iface_arm(uct_iface_h tl_iface, unsigned events)
+{
+    uct_rc_mlx5_iface_common_t *iface =
+            ucs_derived_of(tl_iface, uct_rc_mlx5_iface_common_t);
+    uct_ib_mlx5_md_t *md UCS_V_UNUSED =
+            uct_ib_mlx5_iface_md(&iface->super.super);
+
+    ucs_assert(md->flags & UCT_IB_MLX5_MD_FLAG_DEVX_CQ);
+    return uct_rc_mlx5_iface_devx_arm(iface, events);
+}
+
+static ucs_status_t
+uct_gga_mlx5_iface_query(uct_iface_h tl_iface, uct_iface_attr_t *iface_attr)
+{
+    uct_ib_iface_t *iface = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    ucs_status_t status;
+
+    status = uct_ib_iface_query(iface, UCT_IB_RETH_LEN, iface_attr);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    iface_attr->device_addr_len += ucs_offsetof(uct_gga_mlx5_dev_addr_t,
+                                                ib_addr);
+    iface_attr->ep_addr_len      = sizeof(uct_gga_mlx5_ep_address_t);
+    iface_attr->cap.flags = /*
+                            UCT_IFACE_FLAG_PUT_ZCOPY |
+                            UCT_IFACE_FLAG_GET_ZCOPY |
+                            UCT_IFACE_FLAG_PENDING   | */
+                            UCT_IFACE_FLAG_CONNECT_TO_EP |
+                            UCT_IFACE_FLAG_CB_SYNC;
+
+    iface_attr->cap.event_flags = UCT_IFACE_FLAG_EVENT_SEND_COMP |
+                                  UCT_IFACE_FLAG_EVENT_FD;
+    return UCS_OK;
+}
+
+static ucs_status_t
+uct_gga_mlx5_ep_enable_mmo(uct_gga_mlx5_ep_t *ep)
```

Comment:
IMO everything in this function except init2init devx in/opt/command should be moved out to the ep constructor